lib/volcanic/authenticator/v1/privilege.rb

```diff
@@ -22,13 +22,27 @@ def initialize(scope:, permission_id: nil, group_id: nil, allow:, **args)
         @allow = allow
       end
 
+      def <=>(other)
+        if scope == other.scope
+          allow_bool <=> other.allow_bool
+        else
+          Scope.parse(scope).specificity_score <=> Scope.parse(other.scope).specificity_score
```

Comment:
why is `scope` not an instance of `Scope`? surely that would be more coherent?

also.... `scope <=> other.scope` gives the same value as `scope.specificity_score <=> other.scope.specificity_score` but with better encapsulation - let `Scope` figure out how to compare specificity with another `Scope`